tools/msu_finder.rb

```diff
@@ -0,0 +1,760 @@
+    If you'd like to double check on false positives, you can use the -d flag and manually verify
+    the accuracy of the search results before actually collecting the download links.
+
+    Download:
+    The following trick demonstrates how you can automatically download the updates:
+    ruby #{__FILE__} -q "ms15-100" -r x86 > /tmp/list.txt && wget -i /tmp/list.txt
+
+    Patch Extraction:
+    After downloading the patch, you can use the extract_msu.bat tool to automatically extract
+    Microsoft patches.
+    |
+    end
+
+    def self.get_parsed_options
+      options = {}
+
+      parser = OptionParser.new do |opt|
+        opt.banner = banner.strip.gsub(/^[[:blank:]]{4}/, '')
+        opt.separator ''
+        opt.separator 'Specific options:'
+
+        opt.on('-q', '--query <keyword>', 'Regex keyword to search') do |v|
+          options[:keyword] = v
+        end
+
+        opt.on('-s', '--search-engine <engine>', '(Optional) The type of search engine to use (Technet or Google). Default: Technet') do |v|
+          case v.to_s
+          when /^google$/i
+            options[:search_engine] = :google
+          when /^technet$/i
+            options[:search_engine] = :technet
+          else
+            raise OptionParser::MissingArgument, "Invalid search engine: #{v}"
+          end
+        end
+
+        opt.on('-r', '--regex <string>', '(Optional) Regex download links') do |v|
+          options[:regex] = v
+        end
+
+        opt.on('--apikey <key>', '(Optional) Google API key. Set this if the search engine is Google') do |v|
+          options[:google_api_key] = v
+        end
+
+        opt.on('--cx <id>', '(Optional) Google search engine ID. Set this if the search engine is Google') do |v|
+          options[:google_search_engine_id] = v
+        end
+
+        opt.on('-d', '--dryrun', '(Optional) Perform a search, but do not fetch download links. Default: no') do |v|
+          options[:dryrun] = true
+        end
+
+        opt.on_tail('-h', '--help', 'Show this message') do
+          $stderr.puts opt
+          exit
+        end
+      end
+
+      parser.parse!
+
+      if options.empty?
+        raise OptionParser::MissingArgument, 'No options set, try -h for usage'
+      elsif options[:keyword].nil? || options[:keyword].empty?
+        raise OptionParser::MissingArgument, '-q is required'
+      end
+
+      unless options[:search_engine]
+        options[:search_engine] = :technet
+      end
+
+      if options[:search_engine] == :google
+        if options[:google_api_key].nil? || options[:google_search_engine_id].empty?
+          raise OptionParser::MissingArgument, 'Search engine is Google, but no API key specified'
+        elsif options[:google_search_engine_id].nil? || options[:google_search_engine_id].empty?
+          raise OptionParser::MissingArgument, 'Search engine is Google, but no search engine ID specified'
+        end
+      end
+
+      options
+    end
+  end
+
+  class Module < Base
```

Comment:
I'm not sure if having a class named `Module`, which is an already used word in Ruby core is a good idea (http://ruby-doc.org/core-2.2.0/Module.html)